lightning/src/chain/keysinterface.rs

```diff
@@ -1261,7 +1263,11 @@ impl KeysInterface for KeysManager {
 
 	fn generate_channel_keys_id(&self, _inbound: bool, _channel_value_satoshis: u64, user_channel_id: u128) -> [u8; 32] {
 		let child_idx = self.channel_child_index.fetch_add(1, Ordering::AcqRel);
-		assert!(child_idx <= core::u32::MAX as usize);
+		// child_idx is the only thing guaranteed to make each channel unique without a restart
+		// (though `user_channel_id` should help, depending on user behavior). If it manages to
```

Comment:
To keep it consistent:
```suggestion
		// (though user_channel_id should help, depending on user behavior). If it manages to
```

Reply:
nah, just add the ticks to `child_idx` instead

Reply:
> nah, just add the ticks to `child_idx` instead

I don't care which way to go in a regular comment, as long as it's consistent (if we want ticks, `AtomicUsize` and `u32::MAX` below are also candidates).